modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+namespace Drupal\commerce_promotion\Element;
+
+use Drupal\commerce\Element\CommerceElementBase;
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a form element for embedding the coupon redemption form.
+ *
+ * Usage example:
+ * @code
+ * $form['coupons'] = [
+ *   '#type' => 'commerce_promotion_coupon_redemption_form',
+ *   // An order entity that valid redeemed coupons will be applied to.
```

Comment:
'that valid' doesn't sound valid
"The order to which the coupon will be applied to."